linuxmodule/click-linuxtool.pl

```diff
@@ -340,6 +340,9 @@ ($$)
 	    if ($d eq "rbtree_latch.h") {
 	    	s{container_of\(node, struct latch_tree_node, node\[idx\]\)}{idx==0 ? container_of(node, struct latch_tree_node, node[0]) : container_of(node, struct latch_tree_node, node[1])};
 	    }
+	    if ($d eq "irq.h") {
+	    	s{enum irqchip_irq_state}{int}g;
+	    }
```

Comment:
I fixed this by fixing its type with `enum irqchip_irq_state : int;`.

But that requires C++11. Your fix is probably preferable…